clang/lib/CIR/Dialect/Transforms/TargetLowering/Targets/X86.cpp

```diff
@@ -92,6 +92,21 @@ class X86_64TargetLoweringInfo : public TargetLoweringInfo {
       : TargetLoweringInfo(std::make_unique<X86_64ABIInfo>(LM, AVXLevel)) {
     assert(!::cir::MissingFeatures::swift());
   }
+
+  unsigned getTargetAddrSpaceFromCIRAddrSpace(
+      mlir::cir::AddressSpaceAttr addressSpaceAttr) const override {
+    using Kind = mlir::cir::AddressSpaceAttr::Kind;
+    switch (addressSpaceAttr.getValue()) {
+    case Kind::offload_private:
+    case Kind::offload_local:
+    case Kind::offload_global:
+    case Kind::offload_constant:
+    case Kind::offload_generic:
+      return 0;
+    default:
+      llvm_unreachable("Unknown CIR address space for this target");
```

Comment:
Same as for AArch64; maybe that can even be the default implementation for the virtual method.